Assets/Builder.cs

```diff
@@ -91,9 +91,113 @@ public static Vector2 Perp(Vector2 d)
         return p;
     }
 
-    public static Geometry PolylineToPolygon(Geometry geometry, float extrude)
+    private static Vector2 Miter(Vector3 d0, Vector3 d1, Vector3 n0, Vector3 n1)
     {
-        List<Vector2> geometryPoints = new List<Vector2>(geometry.points.Select(v => new Vector2(v.x, v.y)));
+        var n0n1 = n0 + n1;
+        n0n1.Normalize();
+
+        Vector2 miter = new Vector2();
+
+        double theta = Math.Atan2(d1.y, d1.x) - Math.Atan2(d0.y, d0.x);
+        if (theta < 0.0)
+        {
+            theta += 2.0 * Math.PI;
+        }
+
+        double sinHalfTheta = Math.Sin(theta * 0.5);
+        miter = n0n1 * (float)(1.0f / Math.Max(sinHalfTheta, 1e-5f));
```

Comment:
Can we use `Single.Epsilon` here? https://msdn.microsoft.com/en-us/library/system.single.epsilon(v=vs.110).aspx